doc/modules/clustering.rst

```diff
@@ -1319,7 +1319,7 @@ labels, rename 2 to 3, and get the same score::
 
   >>> labels_pred = [1, 1, 0, 0, 3, 3]
   >>> metrics.rand_score(labels_true, labels_pred)
-  0.66...
+  np.float64(0.66...)
```

Comment:
As a reader, it is a little inconsistent that this has a `np.float64` when two lines lower, `metrics.adjusted_rand_score` does not.

Given this inconsistency, what do you think of using `np.set_printoptions(legacy="1.25")`?

Reply:
Yeah it looks like we don't have any consistency between metrics whether we return a numpy scalar or a Python float ...

I would be of the opinion of showing the Numpy 2 output, because that would be the one that most people will see (as numpy 2 usage grows) when executing the code.

There is also the option of scipy-doctest to have strict checking = False so that `np.float64(3.14)` and `3.14` are compared equal for the doctest see #30496 if you so wish :wink:. But I would rather have this PR merged before #30496.

> Given this inconsistency, what do you think of using np.set_printoptions(legacy="1.25")?

I would rather live in the future :wink:. Not sure, but would `np.set_printoptions(legacy="1.25")` not have other side-effect.

Reply:
> Would np.set_printoptions(legacy="1.25") not have other side-effects?

I was thinking of enabling it just for doc-testing.

As long as we have a plan to make it more consistent, I'm okay with the inconsistency for now.

Reply:
From 1000 feet, and just to keep on the back of our minds in case the need arises in the future: scipy-doctest has a context manager hook to exectute doctests under; numpy/scipy use it to manipulate warnings, but it can also be used for other things like tweaking the numpy formatting for doctests specifically.